### PR TITLE
Graph csr

### PR DESCRIPTION
**Pull request by DvirDukhan**
CSR matrices

### CODE REVIEW

File: deps/GraphBLAS/Config/GraphBLAS.h.in [hunk 1]
```diff
@@ -5557,8 +5557,8 @@ typedef struct
     GrB_Index nvals ;       // Number of none zero values in matrix
     GrB_Index nnz_idx ;     // Index of current none zero value
     int64_t p ;             // Number of none zero values in current column
-    int64_t col_idx ;       // Index of current column
-    GrB_Index ncols ;       // Total number of coulmns in matrix
+    int64_t idx ;           // Index of current row/col
```
Comment (swilly22) [inline deps/GraphBLAS/Config/GraphBLAS.h.in]: row_idx, update comment accordingly.

File: deps/GraphBLAS/Config/GraphBLAS.h.in [hunk 2]
```diff
@@ -5557,8 +5557,8 @@ typedef struct
...
+    GrB_Index n ;           // Total number of rows/col in matrix
```
Comment (swilly22) [inline deps/GraphBLAS/Config/GraphBLAS.h.in]: GrB_Index nrows ;       // Total number of rows in matrix

File: deps/GraphBLAS/Config/GraphBLAS.h.in [hunk 3]
```diff
@@ -5568,11 +5568,11 @@ GrB_Info GxB_MatrixTupleIter_new
     GrB_Matrix A                // matrix being iterated
 ) ;
 
-// Iterate over a specific column
-GrB_Info GxB_MatrixTupleIter_iterate_column
+//iterate over specific index
```
Comment (swilly22) [inline deps/GraphBLAS/Config/GraphBLAS.h.in]: //iterate over specific row

File: deps/GraphBLAS/Config/GraphBLAS.h.in [hunk 4]
```diff
@@ -5568,11 +5568,11 @@ GrB_Info GxB_MatrixTupleIter_new
...
+GrB_Info GxB_MatrixTupleIter_iterate
```
Comment (swilly22) [inline deps/GraphBLAS/Config/GraphBLAS.h.in]: GxB_MatrixTupleIter_iterate_row

File: deps/GraphBLAS/Config/GraphBLAS.h.in [hunk 5]
```diff
@@ -5568,11 +5568,11 @@ GrB_Info GxB_MatrixTupleIter_new
...
 (
-    GxB_MatrixTupleIter *iter,       // iterator to use
-    GrB_Index colIdx        // column index to iterate over
+    GxB_MatrixTupleIter *iter,
+    GrB_Index idx
```
Comment (swilly22) [inline deps/GraphBLAS/Config/GraphBLAS.h.in]: rowIdx // row index to iterate over

File: deps/GraphBLAS/Config/GraphBLAS.h.in [hunk 6]
```diff
@@ -5568,11 +5568,11 @@ GrB_Info GxB_MatrixTupleIter_new
...
 ) ;
 
-// Iterate over a specific column
-GrB_Info GxB_MatrixTupleIter_iterate_column
+//iterate over specific index
+GrB_Info GxB_MatrixTupleIter_iterate
 (
-    GxB_MatrixTupleIter *iter,       // iterator to use
-    GrB_Index colIdx        // column index to iterate over
+    GxB_MatrixTupleIter *iter,
```
Comment (swilly22) [inline deps/GraphBLAS/Config/GraphBLAS.h.in]: GxB_MatrixTupleIter *iter, // iterator to use

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 7]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
 {
     GB_WHERE ("GxB_MatrixTupleIter_new (A)") ;
     GB_RETURN_IF_NULL_OR_FAULTY (A) ;
-    
-    GrB_Index ncols ;
-    GrB_Matrix_ncols (&ncols, A) ;
+
+    GrB_Index n;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: GrB_Index nrows;

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 8]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
...
+    GrB_Matrix_nrows(&n, A);
 
     *iter = NULL ;
     GB_MALLOC_MEMORY (*iter, 1, sizeof (GxB_MatrixTupleIter)) ;
     GrB_Matrix_nvals (&((*iter)->nvals), A) ;
     (*iter)->A = A ;
     (*iter)->nnz_idx = 0 ;
-    (*iter)->col_idx = 0 ;
-    (*iter)->ncols = ncols ;
+    (*iter)->idx = 0 ;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: row_idx

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 9]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
...
+    (*iter)->n = n;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: nrows = nrows

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 10]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
...
     (*iter)->p = A->p[0] ;
     return (GrB_SUCCESS) ;
 }
 
-// Update iter to scan specific column
-GrB_Info GxB_MatrixTupleIter_iterate_column
+GrB_Info GxB_MatrixTupleIter_iterate
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: GxB_MatrixTupleIter_iterate_row

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 11]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
...
 (
     GxB_MatrixTupleIter *iter,
-    GrB_Index colIdx
+    GrB_Index idx
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: rowIdx

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 12]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
...
 )
 {
-    GB_WHERE ("GxB_MatrixTupleIter_iterate_column (iter, colIdx)") ;
-    GB_RETURN_IF_NULL (iter) ;
-    
-    if (colIdx < 0 && colIdx >= iter->ncols) {
-        return (GB_ERROR (GrB_INVALID_INDEX, (GB_LOG, "Column index out of range"))) ;
+    GB_WHERE("GxB_MatrixTupleIter_iterate (iter, idx)");
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: GxB_MatrixTupleIter_iterate_row (iter, rowIdx)

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 13]
```diff
@@ -13,40 +13,40 @@ GrB_Info GxB_MatrixTupleIter_new
...
+    GB_RETURN_IF_NULL(iter);
+
+    if (idx < 0 && idx >= iter->n)
+    {
+        return (GB_ERROR(GrB_INVALID_INDEX, (GB_LOG, "Index out of range")));
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: Row index...

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 14]
```diff
@@ -71,30 +71,34 @@ GrB_Info GxB_MatrixTupleIter_next
...
 
     //--------------------------------------------------------------------------
-    // extract the column indices
+    // extract the row indices
     //--------------------------------------------------------------------------
 
-    const int64_t *Ap = A->p ;
-    int64_t j = iter->col_idx ;
+    const int64_t *Ap = A->p;
+    int64_t i = iter->idx;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: iter->row_idx

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 15]
```diff
@@ -71,30 +71,34 @@ GrB_Info GxB_MatrixTupleIter_next
...
 
-    for (; j < iter->ncols ; j++)
+    for (; i < iter->n; i++)
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: iter->nrows

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 16]
```diff
@@ -110,7 +114,7 @@ GrB_Info GxB_MatrixTupleIter_reset
     GB_WHERE ("GxB_MatrixTupleIter_reset (iter)") ;
     GB_RETURN_IF_NULL (iter) ;
     iter->nnz_idx = 0 ;
-    iter->col_idx = 0 ;
+    iter->idx = 0 ;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: row_idx

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 17]
```diff
@@ -125,12 +129,12 @@ GrB_Info GxB_MatrixTupleIter_reuse
     GB_WHERE ("GxB_MatrixTupleIter_reuse (iter, A)") ;
     GB_RETURN_IF_NULL (iter) ;
     GB_RETURN_IF_NULL_OR_FAULTY (A) ;
-    
-    GrB_Index ncols ;
-    GrB_Matrix_ncols (&ncols, A) ;
+
+    GrB_Index n;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: nrows

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c [hunk 18]
```diff
@@ -125,12 +129,12 @@ GrB_Info GxB_MatrixTupleIter_reuse
...
+    GrB_Matrix_nrows(&n, A);
 
     iter->A = A ;
-    iter->ncols = ncols ;
+    iter->n = n ;
```
Comment (swilly22) [inline deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: nrows

File: src/graph/graph.c [hunk 19]
```diff
@@ -568,7 +569,8 @@ void Graph_GetNodeEdges(const Graph *g, const Node *n, GRAPH_EDGE_DIR dir, int e
 
         while(true) {
             bool depleted = false;
-            GxB_MatrixTupleIter_next(tupleIter, &srcNodeID, NULL, &depleted);
+            // GxB_MatrixTupleIter_next(tupleIter, &srcNodeID, NULL, &depleted);
```
Comment (swilly22) [inline src/graph/graph.c]: Remove comment.

File: deps/GraphBLAS/Config/GraphBLAS.h.in (hunk 1 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Config/GraphBLAS.h.in]: done

File: deps/GraphBLAS/Config/GraphBLAS.h.in (hunk 2 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Config/GraphBLAS.h.in]: done

File: deps/GraphBLAS/Config/GraphBLAS.h.in (hunk 3 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Config/GraphBLAS.h.in]: done

File: deps/GraphBLAS/Config/GraphBLAS.h.in (hunk 4 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Config/GraphBLAS.h.in]: done

File: deps/GraphBLAS/Config/GraphBLAS.h.in (hunk 6 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Config/GraphBLAS.h.in]: done

File: deps/GraphBLAS/Config/GraphBLAS.h.in (hunk 5 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Config/GraphBLAS.h.in]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 7 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 8 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 9 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 10 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 11 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 12 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 13 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 14 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 15 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 16 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 17 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c (hunk 18 above)
Comment (DvirDukhan) [reply deps/GraphBLAS/Source/GxB_Matrix_tuple_iter.c]: done

File: src/graph/graph.c (hunk 19 above)
Comment (DvirDukhan) [reply src/graph/graph.c]: done

**Pull request merged**